### ISSUE TITLE
`no-emphasis-as-heading` (`MD036`) - false positive on indented block & conflicts with `heading-start-left` (`MD023`)

### ISSUE THREAD

**Issue by DetachHead** [rvben/rumdl#721]
### rumdl version

0.2.31

### Installation method

Other

### Bug description

indented blocks such as mkdocs admonitions incorrectly trigger `no-emphasis-as-heading`. in my codebase, we have some emphasis headings inside admonitions, since we don't want them showing up in the TOC (and to avoid unexpected ambiguous behavior in general)

### Steps to reproduce

1. create the following `.md` file:
    ```md
    !!! note
    
        _Foo_
    ```
2. run `rumdl check` on it (`no-emphasis-as-heading` error)
3. change the contents of the file to:
    ```md
    !!! note
    
        # Foo
    ```
4. run `rumdl check` again (`heading-start-left` error)

### Expected behavior

no error

### Actual behavior

```
[MD036] Emphasis used instead of a heading: 'Foo'
```

### Configuration

```toml
[global]
flavor = "mkdocs"
enable = ["MD036", "MD023"]
```

### Operating system

Linux

**Comment by rvben** [comment]
Fixed in v0.2.32. MkDocs flavor detection clears the code-block flag for admonition and content tab bodies, which removed MD036's only guard there, and MD023 never had one. Both rules now skip admonition and content tab bodies under the mkdocs flavor. This also removes the destructive interaction where MD036's opt-in autofix plus MD023's fix pulled a heading out of the admonition in a single `--fix` pass. Thanks for reporting both symptoms together, the conflict framing is what surfaced the fix cascade.
